tests/validation/tests/v3_api/test_node_template.py

```diff
@@ -0,0 +1,283 @@
+        admin_client, node, rancherKubernetesEngineConfig)
+    cluster_cleanup(admin_client, cluster)
+
+##############################################################################
+
+@if_test_rbac
+def test_rbac_cloud_credential_create(remove_resource):
+    client = get_user_client()
+    cloud_credential = cloud_credential_do(client)
+    credentials = client.list_cloud_credential(name=cloud_credential.name)
+    assert len(credentials) == 1
+    time.sleep(.5)
+    remove_resource(cloud_credential)
+
+@if_test_rbac
+def test_rbac_cloud_credential_list(remove_resource):
+    client = get_user_client()
+    cloud_credential = cloud_credential_do(client)
+    credentials = client.list_cloud_credential(name=cloud_credential.name)
+    assert len(credentials) == 1
+    # Admin should be able to list credential
+    admin_client = get_admin_client()
+    credentials = admin_client.list_cloud_credential(name=cloud_credential.name)
+    assert len(credentials) == 1
+    # User 2 should not be able to list credential
+    user1, user1_token = create_user(get_admin_client())
+    user1_client = get_client_for_token(user1_token)
+    credentials2 = user1_client.list_cloud_credential(name=cloud_credential.name)
+    assert len(credentials2) == 0
+    time.sleep(.5)
+    remove_resource(cloud_credential)
+
+
+@if_test_rbac
+def test_rbac_cloud_credential_delete(remove_resource):
+    client = get_user_client()
+    cloud_credential = cloud_credential_do(client)
+    # User1 should be able to delete own credential
+    client.delete(cloud_credential)
+    credentials = client.list_cloud_credential(name=cloud_credential.name)
+    assert len(credentials) == 0
+    # Admin should be able to delete credential created by user1
+    admin_client = get_admin_client()
+    cloud_credential2 = cloud_credential_do(client)
+    admin_client.delete(cloud_credential2)
+    credentials = client.list_cloud_credential(name=cloud_credential2.name)
+    assert len(credentials) == 0
+    # User 2 should not be able to delete credential created by user1
+    user1, user1_token = create_user(get_admin_client())
+    user1_client = get_client_for_token(user1_token)
+    cloud_credential3 = cloud_credential_do(client)
+    with pytest.raises(ApiError) as e:
+            user1_client.delete(cloud_credential3)
+    assert e.value.error.status == 403
+    time.sleep(.5)
+    remove_resource(cloud_credential3)
+
+
+@if_test_rbac
+def test_rbac_cloud_credential_edit(remove_resource):
+    client = get_user_client()
+    cloud_credential = cloud_credential_do(client)
+    # User1 should be able to edit own credential
+    do_cloud_credential_config = {"accessToken": "12345"}
+    client.update(cloud_credential,
+                  digitaloceancredentialConfig=do_cloud_credential_config)
+    # Admin should be able to edit credential created by user1
+    admin_client = get_admin_client()
+    do_cloud_credential_config = {"accessToken": "678910"}
```

Comment:
Can an admin update it using an invalid token? i think we get some error here.

Reply:
changed edit from token to name